repositories/professor_repository.go

```diff
@@ -0,0 +1,126 @@
+package repositories
+
+import (
+	"class-review-backend/models"
+	"log"
+
+	"github.com/jmoiron/sqlx"
+)
+
+type IProfessorRepository interface {
+	GetAllProfessors() (*[]models.Professor, error)
+	GetProfessorByUniqname(string)(*models.Professor, error)
+	PostProfessor(*models.Professor) error
+	UpdateProfessor(*models.Professor, string) (*models.Professor, error)
+	DeleteProfessor(string) error
+	GetProfessorStats() (*[]models.ProfessorStats, error)
+	GetProfessorStatsByUniqname(string) (*models.ProfessorStats, error)
+}
+
+// Implements IProfessorRepository
+type ProfessorRepository struct {
+	Database *sqlx.DB
+}
+
+func DefaultProfessorRepository(db *sqlx.DB) *ProfessorRepository {
+	return &ProfessorRepository{
+		Database: db,
+	}
+}
+
+func (pr *ProfessorRepository) GetAllProfessors() (*[]models.Professor, error) {
+	var professors []models.Professor
+	err := pr.Database.Select(&professors, `SELECT professor_uniqname, 
+ 												   professor_name 
+										    FROM professors`)
+	if err != nil {
+		log.Println("Error in GetAllProfessors:", err)
+		return &professors, err
+	}
+
+	return &professors, nil
+}
+
+func (pr *ProfessorRepository) GetProfessorByUniqname(uniqname string) (*models.Professor, error) {
+	var professor models.Professor
+	err := pr.Database.Get(&professor, `SELECT professor_uniqname, 
+											   professor_name 
+									 	FROM professors WHERE professor_uniqname=?`, uniqname)
+	if err != nil {
+		log.Println("Error in GetProfessorByUniqname:", err)
+		return &professor, err
+	}
+
+	return &professor, nil
+}
+
+func (pr *ProfessorRepository) PostProfessor(professorInput *models.Professor) error {
+	_, err := pr.Database.NamedExec(`INSERT INTO professors 
+									 VALUES (:professor_uniqname, 
+											 :professor_name)`, *professorInput)
+	if err != nil {
+		log.Println("Error in PostProfessor:", err)
+		return err
+	}
+	return nil
+}
+
+func (pr *ProfessorRepository) UpdateProfessor(professorInput *models.Professor, uniqname string) (*models.Professor, error) {
+	professorInput.Uniqname = uniqname
+	_, err := pr.Database.NamedExec(`UPDATE professors 
+									 SET professor_name = :professor_name
+									 WHERE professor_uniqname = :professor_uniqname`, *professorInput)
```

Comment:
I believe NamedExec expects a pointer to a struct as its second argument so that it can modify the struct it's given